### PR TITLE
8278068: Fix next-line modifier (snippet markup)

### PR DESCRIPTION
**Pull request by pavelrappo**
This PR fixes three interrelated issues:

  * https://bugs.openjdk.java.net/browse/JDK-8277027
  * https://bugs.openjdk.java.net/browse/JDK-8278068
  * An accidentally found bug in markup diagnostics

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issues
 * [JDK-8278068](https://bugs.openjdk.java.net/browse/JDK-8278068): Fix next-line modifier (snippet markup)
 * [JDK-8277027](https://bugs.openjdk.java.net/browse/JDK-8277027): Treat unrecognized markup as snippet text, but warn about it

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6739/head:pull/6739` \
`$ git checkout pull/6739`

Update a local copy of the PR: \
`$ git checkout pull/6739` \
`$ git pull https://git.openjdk.java.net/jdk pull/6739/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6739`

View PR using the GUI difftool: \
`$ git pr show -t 6739`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6739.diff">https://git.openjdk.java.net/jdk/pull/6739.diff</a>

</details>

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/SnippetTaglet.java [hunk 1]
```diff
@@ -363,12 +371,16 @@ private static String diff(String inline, String external) {
...
 
-    private StyledText parse(Resources resources, Optional<Language> language, String content) throws ParseException {
-        Parser.Result result = new Parser(resources).parse(language, content);
+    private StyledText parse(Resources resources, Diag diag, Optional<Language> language, String content) throws ParseException {
+        Parser.Result result = new Parser(resources).parse(diag, language, content);
         result.actions().forEach(Action::perform);
         return result.text();
     }
 
+    public interface Diag {
```
[the remark this answers is not in the document]
Comment (pavelrappo) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/SnippetTaglet.java]: I'll go with Diags, thank you. In my mind, "factory" is reserved for a literal factory or an object which is modelled as per the "Abstract Factory" creational design pattern from the GoF book.

**Review by jonathan-gibbons — APPROVED** [review]
I like the simplicity of the fix ;-)

**Pull request closed without merging**